runtime-light/stdlib/crypto/crypto-functions.cpp

```diff
@@ -343,3 +348,100 @@ task_t<Optional<string>> f$openssl_decrypt(string data, const string &method, co
   }
   co_return string{response.inner.value.data(), static_cast<string::size_type>(response.inner.value.size())};
 }
+
+namespace {
+constexpr std::array<std::pair<const char *, tl::HashAlgorithm>, 6> HASH_ALGOS = {{{"md5", tl::HashAlgorithm::MD5},
+                                                                                   {"sha1", tl::HashAlgorithm::SHA1},
+                                                                                   {"sha224", tl::HashAlgorithm::SHA224},
+                                                                                   {"sha256", tl::HashAlgorithm::SHA256},
+                                                                                   {"sha384", tl::HashAlgorithm::SHA384},
+                                                                                   {"sha512", tl::HashAlgorithm::SHA512}}};
+
+std::optional<tl::HashAlgorithm> parse_hash_algorithm(string algo_str) noexcept {
+  std::string_view algo_sv{algo_str.c_str(), algo_str.size()};
+
+  const auto ichar_equals = [](char a, char b) { return std::tolower(a) == std::tolower(b); };
+
+  const auto *it = std::find_if(std::begin(HASH_ALGOS), std::end(HASH_ALGOS),
+                                [&](auto pair) { return std::ranges::equal(algo_sv, std::string_view{pair.first}, ichar_equals); });
+
+  if (it == std::end(HASH_ALGOS)) {
+    return std::nullopt;
+  }
+  return it->second;
+}
+
+task_t<string> send_and_get_string(tl::TLBuffer &&buffer, bool raw_output) {
```

Comment:
Is there a reason for rvalue reference?

Reply:
Yes, it's semantically borrows buffer.

Reply:
The only outcome of your decision is that you force everyone to write `std::move(buffer)`. Nothing else happens

Reply:
Fixed